src/net.h

```diff
@@ -807,17 +808,13 @@ class CNodeRef
     void AddRef()
     {
         if (_pnode)
-        {
-            LOCK(cs_vNodes);
             _pnode->AddRef();
-        }
     }
 
     void Release()
     {
         if (_pnode)
         {
-            LOCK(cs_vNodes);
             _pnode->Release();
```

Comment:
without the lock, you need to:
tmp = _pnode;
_pnode = nullptr;
tmp->Release();

to avoid misuse in another thread if a context switch occurs between line 821 and 822